thenewboston_node/business_logic/storages/file_system.py

```diff
@@ -173,32 +187,31 @@ def _compress(self, absolute_file_path: Path) -> Path:
         with open(absolute_file_path, 'rb') as fo:
             original_data = fo.read()
 
-        logger.debug('File %s size: %s bytes', absolute_file_path, len(original_data))
-        best_absolute_file_path = absolute_file_path
-        best_data = original_data
+        original_size = len(original_data)
+        logger.debug('File %s size: %s bytes', absolute_file_path, original_size)
+        best_compressed_meta = min(self._yield_compressed_meta(original_data, compressors), key=lambda x: x.size)
+        if self.always_compress or best_compressed_meta.size < original_size:
+            compressed_file_path = Path(str(absolute_file_path) + '.' + best_compressed_meta.compressor)
+            logger.debug('Writing compressed file: %s (%s bytes)', compressed_file_path, len(best_compressed_meta))
+            self._write_file(compressed_file_path, best_compressed_meta.data, mode='wb')
+
+            logger.debug('Removing %s', absolute_file_path)
+            os.remove(absolute_file_path)
+            return compressed_file_path
+
+        return absolute_file_path
 
-        for compressor in self.compressors:
+    @staticmethod
+    def _yield_compressed_meta(original_data, compressors) -> Generator[CompressedDataMeta, None, None]:
```

Comment:
compressing data just to yield compressed meta does not seem right to me. it should be either a better name or something is very wrong here